feat: add accessor protocol support and refactor `stats/base/nanvariancech` (#6167)
stdlib-js/stdlib: lib/node_modules/@stdlib/stats/base/nanvariancech/docs/repl.txt

{{alias}}( N, correction, x, strideX )
    Computes the variance of a strided array ignoring `NaN` values and using a
    one-pass trial mean algorithm.

>     The `N` and stride parameters determine which elements in the strided arrays

Reviewer: ```suggestion
    The `N` and stride parameters determine which elements in the strided array
```
there's only 1 array here

    --------
    // Standard Usage:
    > var x = [ 1.0, -2.0, NaN, 2.0 ];

>     > {{alias}}( 4, 1, x, 1 )

Reviewer 2: ```suggestion
    > {{alias}}( x.length, 1, x, 1 )
```

    ~4.3333

>     // Using `N` and `stride` parameters:

Reviewer 2: ```suggestion
    // Using `N` and stride parameters:
```

    --------
    // Standard Usage:
    > var x = [ 1.0, -2.0, NaN, 2.0 ];
    > {{alias}}( 4, 1, x, 1 )
    ~4.3333

    // Using `N` and `stride` parameters:

>     > x = [ -2.0, 1.0, 1.0, -5.0, 2.0, -1.0 ];

Reviewer: Add a `NaN` element in the array and traverse over it to show that we're ignoring it

    > {{alias}}( 3, 1, x, 2 )
    ~4.3333

    // Using view offsets:

>     > var x0 = new {{alias:@stdlib/array/float64}}( [ 1.0, -2.0, 3.0, 2.0, 5.0, -1.0 ] );

Reviewer: similar comment regarding `NaN` values

    > var x1 = new {{alias:@stdlib/array/float64}}( x0.buffer, x0.BYTES_PER_ELEMENT*1 );
    > {{alias}}( 3, 1, x1, 2 )
    ~4.3333

{{alias}}.ndarray( N, correction, x, strideX, offsetX )
    Computes the variance of a strided array ignoring `NaN` values and using a
    one-pass trial mean algorithm and alternative indexing semantics.

    While typed array views mandate a view offset based on the underlying

>     buffer, the offset parameters support indexing semantics based on a

Reviewer: ```suggestion
    buffer, the offset parameter supports indexing semantics based on a
```

>     starting indices.

Reviewer: ```suggestion
    starting index.
```

    --------
    // Standard Usage:
    > var x = [ 1.0, -2.0, NaN, 2.0 ];

>     > {{alias}}.ndarray( 4, 1, x, 1, 0 )

Reviewer 2: ```suggestion
    > {{alias}}.ndarray( x.length, 1, x, 1, 0 )
```

    ~4.3333

    // Using offset parameter:

>     > var x = [ 1.0, -2.0, 3.0, 2.0, 5.0, -1.0 ];

Reviewer: similar comment regarding `NaN` values

    ~4.3333

    // Using offset parameter:

>     > var x = [ 1.0, -2.0, 3.0, 2.0, 5.0, -1.0, NaN, NaN ];

Reviewer 3: ```suggestion
    > x = [ 1.0, -2.0, 3.0, 2.0, 5.0, -1.0, NaN, NaN ];
```
